freqtrade/optimize/interface.py

```diff
@@ -0,0 +1,59 @@
+"""
+IHyperOpt interface
+This module defines the interface to apply for hyperopts
+"""
+
+from abc import ABC, abstractmethod
+from typing import Dict, Any, Callable
+
+from pandas import DataFrame
+
+
+class IHyperOpt(ABC):
```

Comment:
The organization of this folder is as if hyperopt dominates everything else, in reality there are other unrelated modules (BT, Edge). Interface file is placed in optimize which can be interpreted as an interface to all modules in optimize. Same goes for below file (resolver). I am not averse to let it go for this PR knowing that changing the folder structure could implicate another round of refactoring. But IMO folders should be created for each module and its assets or at least call the file `interface_hyperopt.py`. It can be on a followup issue though ...

Reply:
good point, will rename them to `hyperopt_*` 
i don't see much sense in moving them to another module tbh as 
a) hyperoptimization IS optimization
b) `Hyperopt` is a subclass of backtesting (for whatever reason ...)

Reply:
Wait ... 
Your points are bloody valid. tbh I've never even looked into the code of hyperopt. after reading a bit I thing that the interface of BT, Edge, Hyperopt can be the same ... they are all using backtesting data in order to optimize. 
Please give me two more days on that :D

Reply:
the interface is kinda the same - that's why it's subclassing Backtesting (which is however not ideal, but a pain to remove and currently not worth the effort)